src/commands/git/revert.ts

```diff
@@ -71,8 +76,42 @@ export class RevertGitCommand extends QuickCommand<State> {
 		return false;
 	}
 
-	execute(state: RevertStepState<State<GitRevisionReference[]>>) {
-		state.repo.revert(...state.flags, ...state.references.map(c => c.ref).reverse());
+	async execute(state: RevertStepState<State<GitRevisionReference[]>>) {
+		for (const ref of state.references.reverse()) {
+			try {
+				await state.repo.git.revert(ref.ref, state.flags);
+			} catch (ex) {
+				if (ex instanceof RevertError) {
+					let shouldRetry = false;
+					if (ex.reason === RevertErrorReason.LocalChangesWouldBeOverwritten) {
+						const response = await showShouldCommitOrStashPrompt();
+						if (response === 'Stash') {
+							await executeCommand(Commands.GitCommandsStashPush);
+							shouldRetry = true;
+						} else if (response === 'Commit') {
+							await executeCoreCommand('workbench.view.scm');
+							shouldRetry = true;
+						} else {
+							continue;
+						}
+					}
+
+					if (shouldRetry) {
+						try {
+							await state.repo.git.revert(ref.ref, state.flags);
+						} catch (ex) {
+							Logger.error(ex, this.title);
+							void showGenericErrorMessage(ex.message);
+						}
+					}
+
+					continue;
+				}
+
+				Logger.error(ex, this.title);
+				void showGenericErrorMessage(ex.message);
+			}
+		}
```

Comment:
This is probably going to result in me receiving a crash course on git revert flows, but for my own education, can you explain what this is doing?

Reply:
This will probably change after the latest conversation /w Eric about integrating the auto stash flag to the command. The revert command could return a "changes would be overwritten" error when the user had WIP while reverting. With this piece of code, we would catch that, let the user know they have WIP they should either commit or stash, and ask them what to do. Then, depending on the option they chose, we run & wait for the command and retry the revert action (unless they cancelled).

Since we're adding the auto stash flag, I don't think this specific error will happen again.